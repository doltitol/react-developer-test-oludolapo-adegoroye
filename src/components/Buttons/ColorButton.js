import React, { PureComponent } from 'react';

export class ColorButton extends PureComponent {
  render() {
    return (
      <button
        className='color-button'
        style={{
          border: this.props.active ? '2px solid #5ECE7B' : 'none',
          width: this.props.small ? '18px' : '40px',
          height: this.props.small ? '18px' : '40px',
        }}
        onClick={this.props.onClick}
        data-testid='color-button'
      >
        <div
          style={{
            background: this.props.color,
            width: this.props.small ? '15px' : '35px',
            height: this.props.small ? '15px' : '35px',
          }}
        ></div>
      </button>
    );
  }
}

export default ColorButton;
