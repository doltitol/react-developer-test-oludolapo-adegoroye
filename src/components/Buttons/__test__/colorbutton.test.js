import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ColorButton from '../ColorButton';

describe('<ColorButton /> spec', () => {
  it('renders the component', () => {
    const container = render(<ColorButton color={'#1D1F22'} active={true} />);
    const getByTestId = container.getByTestId;
    const colorButton = getByTestId('color-button');
    expect(colorButton.children[0]).toHaveStyle('background: #1D1F22');
    expect(colorButton).toHaveStyle('border: 2px solid #5ECE7B');
    expect(colorButton).toHaveStyle('width: 40px');
    expect(container.children).toMatchSnapshot();
  });
});
