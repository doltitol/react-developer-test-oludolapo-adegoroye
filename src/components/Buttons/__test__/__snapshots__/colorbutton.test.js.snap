// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AttributeButton /> spec renders the component 1`] = `undefined`;

exports[`<ColorButton /> spec renders the component 1`] = `undefined`;
